vllm/entrypoints/openai/serving_completion.py

```diff
@@ -151,20 +151,16 @@ async def create_completion(
                     log_tracing_disabled_warning()
 
                 if isinstance(sampling_params, BeamSearchParams):
-                    if isinstance(self.engine_client, AsyncLLMEngine):
-                        generator = self.engine_client.beam_search(
-                            prompt_inputs["prompt_token_ids"], request_id_item,
-                            sampling_params)
-                    elif isinstance(self.engine_client, MQLLMEngineClient):
-                        generator = self.engine_client.beam_search(
-                            prompt_inputs["prompt_token_ids"], request_id_item,
-                            sampling_params, lora_request)
-                    else:
-                        raise ValueError(
-                            "Beam search in the API server is only supported"
-                            " with AsyncLLMEngine and MQLLMEngineClient."
-                            " please add `--disable-frontend-multiprocessing`"
-                            " to use beam search.")
+                    assert isinstance(self.engine_client,
+                                    (AsyncLLMEngine,
```

Comment:
On second thought, you don't actually need the `assert`

You can `beam_search` to the `EngineClientProtocol`

Reply:
I don't think the base `EngineClientProtocol` has a `beam_search` function

Reply:
It does not. But the `EngineClientProtocol` defines the behavior of `AsyncLLMEngine` and `MQLLMEngine`. So now that both support it, you can expand `EngineClientProtocol` to include the `beam_search` api

Reply:
I cannot find `EngineClientProtocol` . does it exist now? @robertgshaw2-neuralmagic

Reply:
I think it's [this](https://github.com/vllm-project/vllm/blob/e0dbdb013dfe5cdbe044317b4d7d55644d6399b3/vllm/engine/protocol.py#L17)?

Reply:
Yep

Reply:
It’s EngineClient(Protocol)

my bad @youkaichao

Reply:
okay, it should be `EngineClient` class. but `MQLLMEngineClient` does not inherit from `EngineClient` . we can make it a future step to absorb beam search implementation into the `EngineClient` .

Reply:
sounds good. I'll go ahead and merge this pr after it is ready. and after you make MQLLMEngine inherit from EngineClient, we can merge separate beam search implementation in one place.

Reply:
Given that it's currently a `Protocol`, `MQLLMEngine` technically doesn't need to subclass it directly. But it would probably be good to anyhow and actually we could consider changing it to an ABC instead.

I agree with @robertgshaw2-neuralmagic  that this method should just be added to `EngineClient` though and we should not need these type assertions. 

Not directly related to this PR but I also think we should consider renaming it to something like `AsyncEngineClient`, and have a way to obtain an instance of an `AsyncEngineClient` which doesn't involve explicit construction. And that would replace explicit use of `AsyncLLMEngine`.

Reply:
changes are welcome on this part!